pkg/cache/tas_flavor_snapshot.go

```diff
@@ -518,8 +518,13 @@ func (s *TASFlavorSnapshot) findLevelWithFitDomains(levelIdx int, required bool,
 	}
 	levelDomains := slices.Collect(maps.Values(domains))
 	sortedDomain := s.sortedDomains(levelDomains)
+	if useLeastFreeCapacityAlgorithm(unconstrained) {
+		// start from the domain with the least amount of free resources
+		slices.Reverse(sortedDomain)
```

Comment:
Would it be possible to return the domains the the reverse order already from `sortedDomains`?

Reply:
Absolutely, done